src/ultk/language/grammar.py

```diff
@@ -558,7 +561,12 @@ def from_module(cls, module_name: str) -> "Grammar":
         """
         module = import_module(module_name)
         grammar = cls(None)
-        for name, value in inspect.getmembers(module):
+        possible_rules = ()
+        if hasattr(module, "grammar_rules") and type(module.grammar_rules) == tuple:
+            possible_rules = module.grammar_rules
+        else:
+            possible_rules = [k for _, k in inspect.getmembers(module)]
```

Comment:
If you want to force this to be a tuple in general, do `tuple(value for _, value in ...)` (also using `value` instead of `k` for a bit more informative naming).  And note the lack of list construction first in the call to `tuple`